examples/soft-kmeans/README.md

```diff
@@ -0,0 +1,108 @@
+Each data point is associated with a log unnormalized probability distribution for each cluster, which is obtained from transformation of the samples of the means (called the "soft Z" transformed samples).
+These soft Z distributions, after softmax normalization, provide the probability of each point to belong to each cluster.
+The distributions of the soft Z samples for 3 individual data points are shown below (one point located on the left, one in the middle, and one on the right).
+
+<img alt="Soft_z analysis" src="./images/soft-z_samples_sc2.png"/>
+
+The leftward point at `x = -0.687`, shows a distribution of higher soft Z values for the cluster 2, that is clearly distinct for the distribution of soft Z values for cluster 1.
+Therefore, it can be unequivocally assigned to the cluster 2.
+Similarly, the rightward point at `x = 2.369` can be unequivocally assigned to cluster 1.
+However, the middle point at `x = 1.4` displays two distributions of soft Z values that are superimposed.
+This highlights the fact that this point is in an ambiguous area, and cannot be assigned confidently to any of the two cluster.
+Note that this uncertainty is not quantified in the traditional "hard" k-means algorithm, whereas it is available here, through these soft Z samples.
+
+This being said, the soft Z distributions can be summarized to their median value, and consequently, the soft k-means clustering can be converted to a "hard" k-means by assigning each point to its cluster of higher median probability.
+
+<img alt="cluster assignment single chain" src="./images/cluster_assignment_sc_initstate-1-0.png" height="400"/>
+
+In this case, because the lower probability mode of the posterior is sampled, the cluster assignment is rather imperfect.
+The points at the frontier of the 2 clusters (framed in dotted line) were assigned to the left cluster, which is the less probable one.
+
+
+
+## Chainsail sampling
+
+### Sampling
+We also sampled the posterior distribution with Chainsail.
+This time, replica exchange allows the main MCMC chain to not get stuck in one mode, and to sample both the low and the high probability modes.
+
+<img alt="sampling chainsail with initial state = (1, 0)" src="./images/sampling_chainsail_initstate-1-0.png" height="400"/>
+
+The sampled means are shown below.
+The two higher peaks correspond to the samples from the higher posterior mode, and the lower peaks to samples from the lower one.
+Similarly to the samples from the single chain, the means sampled from the lower posterior mode are offset from their expected position, but more importantly, samples from the higher posterior mode match the theoretical cluster centroids much better.
+
+<img alt="sampled means chainsail" src="./images/samples_means_chainsail_initstate-1-0.png" height="400"/>
+
+
+### Cluster assignment
+Similarly, the soft Z samples show bimodal distributions.
+
+<img alt="Soft_z analysis" src="./images/soft-z_samples_chainsail2.png"/>
```

Comment:
Maybe it's just too late in the evening, but I currently have the feeling that this plot is the most convincing result here! Essentially, it shows that, with Chainsail, we capture the fact that there is a certain probability for _both_ component 1 and 2 to have created, e.g., the left-most point. This is absent in the soft-Z histograms for the single chain, and thus sampling with a single chain made us miss out on important information :slightly_smiling_face: If you agree, let's highlight this important result clearly to the reader here.